source/Calamari/Deployment/Conventions/ExtractAzureCloudServicePackageConvention.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.IO;
+using System.IO.Packaging;
+using System.Linq;
+using Calamari.Integration.Azure.CloudServicePackage;
+using Calamari.Integration.Azure.CloudServicePackage.ManifestSchema;
+using Calamari.Integration.FileSystem;
+
+namespace Calamari.Deployment.Conventions
+{
+    public class ExtractAzureCloudServicePackageConvention : IInstallConvention
+    {
+        readonly ICalamariFileSystem fileSystem;
+
+        public ExtractAzureCloudServicePackageConvention(ICalamariFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem;
+        }
+
+        public void Install(RunningDeployment deployment)
+        {
+            if (deployment.Variables.GetFlag(SpecialVariables.Action.Azure.CloudServicePackageExtractionDisabled, false))
+                return;
+
+            Log.Verbose("Extracting cspkg");
+            var packagePath = deployment.Variables.Get(SpecialVariables.Action.Azure.CloudServicePackagePath);
+            using (var package = Package.Open(packagePath, FileMode.Open))
+            {
+                var manifest = AzureCloudServiceConventions.ReadPackageManifest(package);
+                var workingDirectory = deployment.CurrentDirectory;
+
+                ExtractContents(package, manifest, "ServiceDefinition", workingDirectory);
```

Comment:
Should these strings ("ServiceDefinition", "NamedStreams", etc.) be stored somewhere as known strings? They're used a couple of times...